internal/pkg/block/block.go

```diff
@@ -7,20 +7,25 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"os"
+	"syscall"
 
 	utilexec "k8s.io/utils/exec"
 )
 
 const (
 	// lsblkCommand is the command to list block devices.
 	lsblkCommand = "lsblk"
+	blkidCmd     = "blkid"
 )
 
 // Interface defines the methods that block should implement
 //
 //go:generate mockgen -copyright_file ../../../hack/mockgen_copyright.txt -destination=mock_block.go -mock_names=Interface=Mock -package=block -source=block.go Interface
 type Interface interface {
 	GetDevices(ctx context.Context) (*DeviceList, error)
+	IsBlockDevice(path string) (bool, error)
+	IsBlkDevUnformatted(device string) (bool, error)
```

Comment:
I'd rename this to `IsFormatted()` and reverse the logic. Then you'd call it as
`block.IsFormatted()` which seems nicer to use.